azerothcore/azerothcore-wotlk
# feat(Core/Script): Playing music from GameObjects and added holiday zones music script

## CHANGES PROPOSED:
-  Add TC commits:
https://github.com/TrinityCore/TrinityCore/commit/002b0a7a65167d15969d98b41af7f91d222df664
https://github.com/TrinityCore/TrinityCore/commit/963278459a1d9df5aa3c2e1c8b5f5f112f6a370c
https://github.com/TrinityCore/TrinityCore/commit/e3560f3f518ee8a76921b4922292805c03871d4a
https://github.com/TrinityCore/TrinityCore/commit/19f8acdb616909758391d1c1459fe61196531f3b
- Modified selection of events for brewfest to be able to play all events at SHATTRATH


## ISSUES ADDRESSED:
- Closes https://github.com/azerothcore/azerothcore-wotlk/issues/3215



## TESTS PERFORMED:
- Builded with no errors
- For Brewfest event: started brewfest event and went to multiple locations to check music (stayed in each for 5 minutes)
- Pirate Day event: started the event and went to the location at booty bay to check music  (stayed for 5 minutes)
- Darkmoon Fair Music: started all fair events and checked music in all their locations  (stayed in each for 5 minutes)
- MidSummer event: started event and checked some location with horde and alliance character  (stayed in each for 5 minutes)

## HOW TO TEST THE CHANGES:

Brewfest event
1. Go to any brewfest event in any city (outside main entrance)
-- Some locations: 
--- Orgrimar: .go xyz 1278.01 -4398.94 26.31 1
--- Stormwind: .go xyz -9129.9 340.76 93.76 0
2. .event start 24
3. Check music


Pirate Day event
1. Go to the event location: .go xyz -14425 530 25 0
3. .event start 50
4. Check music

Darkmoon Fair Music
1. .event start 3
2. .event start 4
3. .event start 5
4. .go xyz -1570.13 146.31 -7.792 1
5. Check music
6. .go xyz -9545.3 93.82 58.94 0
7. Check music
8. .go xyz -2260 5211.64 -9.98 530
9. Check music

MidSummer event
1. .event start 1
2. Go to summer location for horde and aliance
-- Some locations:
--- For horde: .go xyz 143.54 -4723.56 17.24 1
--- For alliance: .go xyz -9377.34 16.67 62.66 0
--- PS: there are many more locations
3. Check music


## KNOWN ISSUES AND TODO LIST:



## Target branch(es):








 
## How to test AzerothCore PRs
 
When a PR is ready to be tested, it will be marked as **[WAITING TO BE TESTED]**.

You can help by testing PRs and writing your feedback here in the PR's page on GitHub. Follow the instructions here:

http://www.azerothcore.org/wiki/How-to-test-a-PR
